R/executionSettings.R

```diff
@@ -0,0 +1,167 @@
+
+# ExecutionSettings ----
+
+#' @description
+#' An R6 class to define an ExecutionSettings object
+#'
+#' @export
+ExecutionSettings <- R6::R6Class(
+  classname = "ExecutionSettings",
+  public = list(
+    initialize = function(connectionDetails = NULL,
+                          connection = NULL,
+                          cdmDatabaseSchema = NULL,
+                          workDatabaseSchema = NULL,
+                          tempEmulationSchema = NULL,
```

Comment:
`tempEmulationSchema` is normally an `option`

Reply:
could we have it as an argument? snowflake requires it for everything selfishly hope to keep it here 😃

Reply:
yes, keep it as an argument but the current convention across other packages is to use

```
tempEmulationSchema = getOption('sqlRenderTempEmulationSchema')
```

So I'm proposing that this behvaiour is consistent.